sdlib/d/gc/tcache.d

```diff
@@ -88,11 +88,18 @@ public:
 		}
 
 		auto pd = getPageDescriptor(ptr);
-
 		auto e = pd.extent;
-		// Slab is not yet supported
-		if (e !is null && !pd.isSlab() && e.finalizer !is null) {
-			e.finalizer(ptr, e.usedCapacity);
+
+		if (pd.isSlab()) {
+			auto si = SlabAllocInfo(pd, ptr);
+			auto finalizer = si.finalizer;
+			if (finalizer !is null) {
+				finalizer(ptr, si.usedCapacity);
```

Comment:
This will call the finalizer on interior pointers and then crash in the arena code. This is not proper.